synedrion/src/uint/secret_signed.rs

```diff
@@ -178,6 +181,21 @@ where
         )
         .expect("the new bound is valid since the constituent ones were")
     }
+
+    /// Multiplies two numbers and returns a new [`SecretSigned`] of twice the bit-width.
+    pub fn mul_wide(&self, rhs: &SecretSigned<T>) -> SecretSigned<T::Wide> {
+        let abs_value = Secret::init_with(|| {
+            self.abs_value()
+                .expose_secret()
+                .mul_wide(rhs.abs_value().expose_secret())
+        });
+        SecretSigned::new_from_abs(
+            abs_value,
+            self.bound() + rhs.bound(),
+            self.is_negative() ^ rhs.is_negative(),
+        )
+        .expect("the new bound is valid since the constituent ones were")
```

Comment:
```suggestion
        .expect("the new bound is valid since the sum of the constituent bounds fitsin a T::Wide")
```

Reply:
arrgh, I had a typo in my suggestion, hope you didn't take that verbatim. :/

Reply:
No, fixed that too